scripts/Muon/GUI/FrequencyDomainAnalysis/plot_widget/duel_plot_maxent_pane/duel_plot_maxent_pane_model.py

```diff
@@ -0,0 +1,212 @@
+        self.reconstructed_data_name = ""
+        self._selection = ""
+
+    def create_options(self):
+        if self._is_groups:
+            return self.get_group_options()
+        else:
+            return self._raw_model.gen_detector_options()
+        return []
+
+    """
+    Handle reconstructed data
+    """
+    def set_reconstructed_data(self, ws, table_name):
+        # get the indices and groups first
+        self.reconstructed_data_name = ws
+        if table_name:
+            table = retrieve_ws(table_name)
+            for index in range(table.rowCount()):
+                data = table.row(index)
+                self.reconstructed_data[index] = data["Group"]
+
+    def add_reconstructed_data(self, workspaces, indices):
+        first, last = 0, 0
+        if self._is_groups:
+            first, last = self.get_first_and_last_group_by_index()
+            last += 1 # to include the last group
+        else:
+            first, last = self.get_first_and_last_detector_by_index()
+        #plus 1 to include last
+        for key in range(first, last):
+            workspaces += [self.reconstructed_data_name]
+            indices += [key]
+        return workspaces, indices
+
+    """
+    Plotting
+    """
+    @staticmethod
+    def _get_freq_label(workspace_name):
+        label = ''
+        if MAXENT_STR in workspace_name:
+            label = f';{MAXENT_STR}'
+        return label
+
+    def _create_workspace_label(self, workspace_name, index):
+        run = str(get_run_numbers_as_string_from_workspace_name(workspace_name, self.context.data_context.instrument))
+        # maxent
+        if MAXENT_STR in workspace_name:
+            freq_label = self._get_freq_label(workspace_name)
+            if RECONSTRUCTED_SPECTRA in workspace_name and self._is_groups:
+                group = self.reconstructed_data[index]
+                return f"{run} {group} {freq_label}"+RECONSTRUCTED_SPECTRA
+            elif RECONSTRUCTED_SPECTRA in workspace_name:
+                return self._raw_model._create_workspace_label(workspace_name,index)+RECONSTRUCTED_SPECTRA
+            else:
+                return f"{run} Frequency {freq_label}"
+        #actual data
+        if self._is_groups and self.reconstructed_data:
+            group = self.reconstructed_data[index]
+            return f"{run} {group}"
+        elif "raw" in workspace_name:
+            return self._raw_model._create_workspace_label(workspace_name,index)
+        return ""
+
+    def get_workspace_list_and_indices_to_plot(self):
+        workspace_list, indices = [], []
+        if self._is_groups:
+            group_list = self.get_group_list()
+            workspace_list, indices = self._time_group_model.get_workspace_list_and_indices_to_plot(True, "Counts", group_list)
+        elif self._run:
+            workspace_list, indices = self._raw_model.get_workspace_list_and_indices_to_plot(True, "Counts", self._selection, self._run,
```

Comment:
is self._run the detectors? am presuming this is the method from raw_pane_model that needs a string of detectors.

Reply:
run is the run number, the detectors come from the selection